src/seedu/addressbook/commands/FavouriteCommand.java

```diff
@@ -0,0 +1,43 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.person.UniquePersonList.AlreadyFavouritedException;
+
+/**
+ * Favourites a person identified using it's last displayed index from the address book.
+ */
+public class FavouriteCommand extends Command {
+
+    public static final String COMMAND_WORD = "favourite";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Favourites the person identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_FAVOURITE_PERSON_SUCCESS = "Favourited %1$s";
+    public static final String MESSAGE_PERSON_ALREADY_FAVOURITED = "Peron is already favourited";
+
+    public FavouriteCommand(int targetVisibleIndex) {
+        super(targetVisibleIndex);
+    }
+
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.favouritePerson(target);
+            return new CommandResult(String.format(MESSAGE_FAVOURITE_PERSON_SUCCESS, target));
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        } catch (PersonNotFoundException pnfe) {
+            return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+        } catch (AlreadyFavouritedException pnfe) {
+            return new CommandResult(MESSAGE_PERSON_ALREADY_FAVOURITED);
+        }
```

Comment:
Well done improving your user experience by catching for AlreadyFavourited cases